round4/team-3.md

```diff
@@ -0,0 +1,117 @@
+
+If we’re able to get to it, we want to add a notification feature that alerts users when prices of their crypto increase. 
+
+### API, Libraries, and Frameworks
+- React Frontend
+- Vite 
+- Component libraries?
+- MySQL + Node.js + Express? Backend
+- WalletConnect API 
+- Have user log in to their account and get their user id from the log in. Use this info to get trade transactions
+- https://etherscan.io/
+- Viewing wallet transaction history
+- API Available: https://etherscan.io/apis
+- (Would require our own backend / server to restrict usage)
+- https://min-api.cryptocompare.com/
+- API to get current prices of crypto currencies 
+- Coingecko free (30 calls / min)
+
+
+### Data Storage
+- Storing user’s account information and etherscan transactions.
+- AWS RDS?
+- Storing etherscan transitions potentially in the browser caching
+- Internal browser caching, localStorage API (public and non sensitive data), and potential server storage for public and non-sensitive data.
+- Data can also be stored on a (potential?) server
+---
+
+## Project Management
+
+### Collaboration and Task Allocation
+- **Leader:** Ahmed Abdellah
+- **Manager:** Chelsea Pan
+- **Remaining Team Members:** Rayyan Shaik, Marc Vaz
+
+Ahmed Abdellah- has background knowledge in crypto and the technologies needed. 
+Chelsea Pan - Project manager, leading backend development (AWS, writing MySQL queries)
+Rayyan Shaik - frontend?
+Marc Vaz - work across stack (backend and frontend)
+
+We will collaborate using notion/google docs to document and write out our pipeline. We will use github to share and produce code
+
+### Risks and Mitigation
+Currently, our biggest risk is using WalletConnect API to have the user login. We’re not exactly sure how we can track which user is logged in to then track their data. If we can’t figure this out before the due date, we may have the user find their own user id and just input it into the extension. 
+
+---
+
+## Milestones and Timeline
+- **Week 1:** Determine tech stack, all technologies needed (APIs), workflow. Develop barebones MVP. No Backend, no Database. Model with users manually inputting their address and their api key. Write API calls and embed them into backend.js
+- **Week 2:** Begin building the backend. Write out all necessary queries and read all documentation for the APIs needed.Set up AWS RDS and learn how to set up a chrome extension. Determine table schemas, how many tables needed, what info needs to be stored. 
+- **Week 3:** Continue backend development. Begin building the frontend, looking into any UI libraries needed to display tables and graphs for the extension. 
+- **Week 4:**  Connect backend to the frontend. Determine logic like what happens when you click on something, what happens in the backend?
```

Comment:
Apologies for my limited understanding of crypto, but I have a few questions regarding the project. Are you planning to display all cryptocurrencies, including Bitcoin, Ethereum, and others? If so, how will you track all of them? Is there a comprehensive cryptocurrency database that you'll be utilizing? Additionally, will the extension only retrieve data for the cryptocurrencies associated with the user's tokens, or will it encompass all available coins

Another consideration is the usability of the Chrome extension itself.It is  to visualize without a Figma file, I assume that the new window that pops up will simply display the user ID to access their wallet, and then users can close it while the extension remains active in the Chrome browser. However, I'm curious about the security implications. What happens if users close the other tab? Will the Chrome extension still be able to track their data? Ensuring security measures are robust is crucial, especially when dealing with sensitive information like crypto.

Reply:
We're currently aiming to track just one of the cryptocurrencies to make sure our extension is usable. If we have time/bandwidth, we will then display all currencies. Also, we will only display data for the currencies associate with the user's tokens. 
In terms of security, all this information is public but we will keep in mind that this is sensitive information. Right now, we're thinking of having a refresh button so a user has to manually update the extension and then it will update all the prices accordingly.